src/test/java/com/arpnetworking/tsdcore/sinks/DimensionFilteringSinkTest.java

```diff
@@ -86,6 +86,32 @@ public void testInjectExcludeWithout() {
         Mockito.verify(_target).recordAggregateData(dataIncluded);
     }
 
+    @Test
+    public void testWhistlist() {
+        final DimensionFilteringSink sink = new DimensionFilteringSink.Builder()
+                .setName("testWhistlist")
+                .setSink(_target)
+                .setWhitelistDimensions(ImmutableSet.of("foo", "bar"))
+                .build();
+        final PeriodicData dataExcluded = TestBeanFactory.createPeriodicDataBuilder()
+                .setDimensions(ImmutableMap.of("foo", "fooVal", "baz", "bazVal"))
+                .build();
+        sink.recordAggregateData(dataExcluded);
+        Mockito.verify(_target, Mockito.never()).recordAggregateData(Mockito.any(PeriodicData.class));
+
+        final PeriodicData dataIncluded = TestBeanFactory.createPeriodicDataBuilder()
```

Comment:
See my other comment, but line 96 was meant to cover the case whereby a metric is excluded because it has an extra dimension, in this case, `baz` - perhaps I'll change `baz` to something that looks less like `bar`.